lib/iris/experimental/ugrid/__init__.py

```diff
@@ -3656,7 +3657,36 @@ def _build_mesh(cf, mesh_var, file_path):
     attributes = {}
     attr_units = get_attr_units(mesh_var, attributes)
 
-    topology_dimension = mesh_var.topology_dimension
+    if hasattr(mesh_var, "volume_node_connectivity"):
+        expected_topology_dimension = 3
+    elif hasattr(mesh_var, "face_node_connectivity"):
+        expected_topology_dimension = 2
+    elif hasattr(mesh_var, "edge_node_connectivity"):
+        expected_topology_dimension = 1
+    else:
+        # Nodes only.  We aren't sure yet whether this is a valid option.
+        expected_topology_dimension = 0
+
+    if not hasattr(mesh_var, "topology_dimension"):
```

Comment:
Would it not make more sense to set `topology_dimension = expected_topology_dimension` outside of this "if" statement and have this logic only handle warnings? It seems to me that there's currently no case where `topology_dimension` does not end up being equal to `expected_topology_dimension`.

Reply:
Ok, well put.
I will try + make this clearer ... :+1: